app/src/androidTest/java/com/example/campusguide/CurrentLocationAcceptanceTest.kt

```diff
@@ -0,0 +1,72 @@
+package com.example.campusguide
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.matcher.ViewMatchers.assertThat
+import androidx.test.filters.SdkSuppress
+import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.GrantPermissionRule
+import androidx.test.uiautomator.*
+import junit.framework.Assert.assertEquals
+import org.hamcrest.CoreMatchers.notNullValue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+private const val LAUNCH_TIMEOUT = 5000L
+
+@RunWith(AndroidJUnit4ClassRunner::class)
+@SdkSuppress(minSdkVersion = 18)
+class CurrentLocationAcceptanceTest {
+
+    private lateinit var device: UiDevice
+
+    @get:Rule
+    var permissionRule = GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION)
+
+    @Before
+    fun startMainActivityFromHomeScreen() {
+
+        device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+        device.pressHome()
+
+        val launcherPackage: String = device.launcherPackageName
+        assertThat(launcherPackage, notNullValue())
+        device.wait(Until.hasObject(By.pkg(launcherPackage).depth(0)), LAUNCH_TIMEOUT)
+
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        val intent = context.packageManager.getLaunchIntentForPackage("com.example.campusguide")
+        context.startActivity(intent)
+
+        device.wait(Until.hasObject(By.pkg("com.example.campusguide").depth(0)), LAUNCH_TIMEOUT)
+    }
+
+    @Test
+    fun checkPreconditions() {
+        assertThat(device, notNullValue())
+    }
+
+    @Test
+    fun clickCurrentLocationButton() {
+
+        // Wait 5 seconds
+        device.waitForWindowUpdate(null, 5000)
```

Comment:
From what I can tell this will always wait the full 5 seconds.
Is there a way to wait just long enough for the stuff we need?
If not, do we really need to wait 5 seconds, or is 1 second sufficient?

Reply:
Yes there is a way, and it makes the tests much faster. So fast that it's no longer possible to follow along visually.
Comes down to what do we want/value more, lmk what you think

Reply:
I'd say we should make the tests run faster, and we can manually add the delays if a test is failing and we want to follow along.

Reply:
Done